padl/transforms.py

```diff
@@ -672,6 +705,15 @@ def _pd_unpack_argument(self, arg) -> bool:
             return True
         return False
 
+    def _pd_call_transform(self, arg):
```

Comment:
I think we should come up with a better name for this function. I find it a little confusing that `pd_call_transform` then calls `_pd_call_transform`.

Reply:
mmm yes, what about `_pd_run_with_try_if_needed`? not a very nice name either

Reply:
Yes, because `Pipeline`s and `AtomicTransform`s implement different `_pd_call_transform` method. This is because in `Pipeline`s the `try` and the Exception catching are done inside `_pd_call_transform`  and in `Pipeline`s this is inside the `__call__` method, so we can track the item `i` that is failing.

Reply:
@wuhu @sjrl